contracts/src/pkg/XApp.sol

```diff
@@ -3,16 +3,33 @@ pragma solidity ^0.8.12;
 
 import { IOmniPortal } from "../interfaces/IOmniPortal.sol";
 import { XTypes } from "../libraries/XTypes.sol";
+import { ConfLevel } from "../libraries/ConfLevel.sol";
 
 /**
  * @title XApp
  * @notice Base contract for Omni cross-chain applications
  */
 contract XApp {
+    /**
+     * @notice Emitted when the OmniPortal contract address is set
+     */
+    event OmniPortalSet(address omni);
+
+    /**
+     * @notice Emitted when the default confirmation level is set
+     * @param conf  Confirmation level
+     */
+    event DefaultConfLevelSet(uint8 conf);
+
     /**
      * @notice The OmniPortal contract
      */
-    IOmniPortal internal immutable omni;
+    IOmniPortal public omni;
+
+    /**
+     * @notice Default confirmation level for xcalls
+     */
+    uint8 public defaultConfLevel;
```

Comment:
What do you think of setting a default value here already and providing a value in constructor if you desire to override? Have this be the finalized level by default

Reply:
We can't overload the constructor so the ux for default is awkward. And I think it's actually a reasonable thing to make people think about, at least once